Add examples for go-spiffe v2 (#114)
spiffe/go-spiffe: v2/examples/spiffe-watcher/README.md

# X.509 SVID Watcher example

> This example shows how a service can obtain X.509 SVIDs and JWT Bundles from the SPIFFE workload API which are automatically rotated before expiration.

Reviewer: ```suggestion
This example shows how a service can obtain automatically rotated X.509 SVIDs and JWT Bundles from the SPIFFE Workload API.
```

> The first step is to create a workload API client. It implements different methods to query the SPIFFE Workload API. It takes a [workloadapi.ClientOption](../../workloadapi/option.go#L10) to create client.

Reviewer: ```suggestion
The first step is to create a Workload API client. The code assumes it is talking to [SPIRE](https://github.com/spiffe/spire) and uses a [workloadapi.ClientOption](../../workloadapi/option.go#L10) to provide the address to the SPIRE Agent socket.
```

```go
client, err := workloadapi.New(ctx, workloadapi.WithAddr(socketPath))
```

In case `workloadapi.WithAddr` is not provided, the value of `SPIFFE_ENDPOINT_SOCKET` environment variable will be used 

```go
client, err := workloadapi.New(ctx)
```

> The library provides a watcher interface type [workloadapi.X509ContextWatcher](../../workloadapi/client.go#L322) to manipulate responses when SVIDs rotations or errors.

Reviewer: ```suggestion
The library uses a watcher interface  [workloadapi.X509ContextWatcher](../../workloadapi/client.go#L322) to send updates (or errors) to clients.
```

> The watcher for X.509 SVID is used in [workloadapi.WatchX509Context](../../workloadapi/client.go#L151) function.

Reviewer: ```suggestion
```

```go
err = client.WatchX509Context(ctx, &x509Watcher{})
```

> The watcher will be notifier every time an SVID is update or an error occurrs

Reviewer: ```suggestion
The watcher will be notified every time an SVID is updated or an error occurs.
```

> It is possible to watch for JWT Bundles updates. The library provide a watcher interface type [workloadapi.JWTBundleWatcher](../../workloadapi/client.go#L333) to manipulate responses on JWT Bundles rotations or errors.
> 
> The watcher for JWT Bundles is used in  [workloadapi.WatchJWTBundles](../../workloadapi/client.go#L201) 

Reviewer: ```suggestion
It is possible to watch for JWT Bundles updates:
```

```go
err = client.WatchJWTBundles(ctx, &jwtWatcher{})
```

The watcher will be notified every time a JWT Bundle  is updated or an error occurs. 

## Building
Build the spiffe-watcher example:

```bash
cd examples/spiffe-watcher
go build
```

## Running
This example assumes the following preconditions:

> - There are a SPIRE server and agent up and running.

Reviewer: ```suggestion
- There is a SPIRE server and agent up and running.
```
